pandas/core/arrays/sparse/array.py

```diff
@@ -1375,23 +1375,6 @@ def _where(self, mask, value):
         result = type(self)._from_sequence(naive_implementation, dtype=dtype)
         return result
 
-    # ------------------------------------------------------------------------
-    # IO
-    # ------------------------------------------------------------------------
-    def __setstate__(self, state) -> None:
-        """Necessary for making this object picklable"""
-        if isinstance(state, tuple):
-            # Compat for pandas < 0.24.0
-            nd_state, (fill_value, sp_index) = state
-            sparse_values = np.array([])
-            sparse_values.__setstate__(nd_state)
-
-            self._sparse_values = sparse_values
-            self._sparse_index = sp_index
-            self._dtype = SparseDtype(sparse_values.dtype, fill_value)
-        else:
-            self.__dict__.update(state)
```

Comment:
Don't we still need this else branch? Or are we totally dropping support for this?